articles/150_roslaunch.md

```diff
@@ -0,0 +1,364 @@
+---
+layout: default
+title: ROS 2 Launch System
+permalink: articles/roslaunch.html
+abstract:
+  The launch system in ROS is responsible for helping the user describe the configuration of their system and then execute it as described. The configuration of the system includes what programs to run, what arguments to pass them, and ROS specific conventions which make it easy to reuse components throughout the system through configuration. Also, because the launch system is the entity that executes processes, it is also responsible for monitoring the state of processes in the system, then reporting and/or reacting to changes in the state of those processes.
+author: '[William Woodall](https://github.com/wjwwood)'
+published: true
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Context
+
+<div class="alert alert-warning" markdown="1">
+TODO: Fill out some context with respect to ROS 1's `roslaunch` and other similar tools which exist in OS's.
+</div>
+
+## Separation of Concern
+
+The launch system can be considered in parts, separated by concern.
+The coarse breakdown is like so:
+
+- Calling Conventions for Processes and Various Styles of Nodes
+- Reporting System for Event
+- System Description and Static Analysis
+- Execution and Verification of the System Description
+- Testing
+
+The purpose of the following sections is to enumerate what the launch system could interact with or do, but is not the requirements list for the launch system in ROS 2.
+The requirements for the launch system will be enumerated in section below based on what's possible in these sections.
+
+## Calling Conventions
+
+In order for the launch system to execute a described system, it needs to understand how it can achieve the description.
+This is an existing phrase in Computer Science[^calling_convention_wikipedia], but this section is not talking specifically about the compiler defined calling convention, through it is appropriating the term to describe a similar relationship.
+In this case, the phrase "calling conventions" is meant to describe the "interface" or "contract" the launch system has with the entities it is executing and monitoring.
```

Comment:
Might be a good idea to define what an entity is before this section. I assume `entities` means the types of nodes: a process with one node, a process manually composed with multiple nodes, a library containing a dynamically composible node, or all of these but with a lifecycle node.

Reply:
I was hoping this sentence would imply "entities are anything that the launch system executes and/or monitors", rather than me specifically defining entities. I used "entities" because "things" sounded unsophisticated.

Perhaps I could reword it to be "... the launch system has with entities, which are anything it executes or monitors."? If not, do you have a suggestion as how to reword it?

Reply:
Oops I assumed entity was going to be a special term. If it's not going to be used later on then I suggest shortening to "... `the launch system has with anything it executes or monitors.`".

Reply:
3473687